packages/bot/src/stores/toolbar-store.js

```diff
@@ -0,0 +1,283 @@
+import filesaver                        from 'file-saver';
+import {
+    observable,
+    action,
+    computed,
+}                                       from 'mobx';
+import { createErrorAndEmit }           from '../utils/error';
+import { observer as globalObserver }   from '../utils/observer';
+import {
+    cleanUpOnLoad,
+    fixCollapsedBlocks,
+    backwardCompatibility,
+    fixArgumentAttribute,
+    addDomAsBlock,
+}                                       from '../scratch/utils';
+import { translate }                    from '../utils/tools';
+import googleDrive                      from '../utils/integrations/googleDrive';
+
+export default class ToolbarStore {
+    constructor(flyout) {
+        this.flyout = flyout;
+    }
+
+    @observable is_toolbox_open = false;
+    @observable is_saveload_modal_open = false;
+    @observable is_save_modal = true;
+    @observable saveload_type = 'local';
+    @observable file_name = 'Untitled Bot';
+    @observable is_google_drive_connected = false;
+
+    @action.bound onRunClick = () => {
+        // TODO
+    }
+
+    @action.bound onStartClick = () => {
+        this.is_toolbox_open = !this.is_toolbox_open;
```

Comment:
- Why should the toolbox close when the bot is running?
- If it should close, the value of `this.is_toolbox_open` should be set to `false`, if the toolbox was already closed it would be opened and vice versa.

Reply:
okay maybe the naming is confusing. If you look at the design, button that toggle the toolbox has a `Start` text on it. That is why i name it `onStartClick`. I'll change the naming to toggleToolbox